PaymentHighway/SPH.swift

```diff
@@ -226,15 +224,15 @@ open class SPH {
         return formattedCardNumberForProcessing(expirationDate)
     }
     
-    open func formattedExpirationDate(_ expirationDate: String) -> String {
+    open func formattedExpirationDate(_ expirationDate: String, _ lastExpirationDateLength: Int) -> String {
```

Comment:
Is there a reason why `lastExpirationDateLength` is needed for this?

Reply:
Yes this was a state before inside the function that make the function unpredictable. Simple rule same input must return same output always. As before the output was depending of the previous call. why is needed the lastExpirationDateLength? I discover the use case. we need a different behaviour if user input sequence is "1", "11", "11/1" -> output "1", "11/", "11/1" comparing when user is deleting chars "11/1", "11/", "11" -> output "11/1", "11/", "11". pay attention to the "11" if the behaviour is not different then we would not be able to delete the / and delete all the date.
That the reason why the state but not good in my opinion have to be clear in the interface. 
Any proposal?

Reply:
in short we need to manage the different behaviour and understand if user is adding chars to the date or deleting chars. even a boolean flag is enough global state is not the solution in my opinion.